tests/test_helpers.py

```diff
@@ -86,6 +86,34 @@ def my_func(param1: str, param2: str, param3: str='a'):
         assert d_keywords[3] == {'param1': 'hello', 'param2': 'there', 'param3': 'b'}, \
             'build_vset with my_func + param_dict2 fails'
 
+        # list of funcs [my_func, my_func] with [param_dict1, param_dict2]
+        vsets = build_vset("vset", [my_func, my_func], [param_dict1, param_dict2], param3='b')
```

Comment:
It'd be nice if we can support a usage like `build_vset("vset", my_func, [param_dict1, param_dict2], param3='b')` for cases when there's only one function but multiple param dicts. The use case for this is when someone wants to try a subset of the full cartesian product of the param values they're interested in. So `build_vset`'s `func` can either be a singleton or a list of the same length as the `param_dict` list (ignoring `param_dict` when it's `None`). 

I guess we could do the same thing for `param_dict` (i.e. it can be a singleton and work with a `func` list) but seems less important so don't worry about it for now.